### ISSUE TITLE
Issue when training MVDeTr

### ISSUE THREAD

**Issue by luorix1** [hou-yz/MVDeTr#1]
Error while training MVDeTr

Size mismatch within transform module

```
Traceback (most recent call last):
  File "main.py", line 186, in <module>
    main(args)
  File "main.py", line 132, in main
    train_loss = trainer.train(epoch, train_loader, optimizer, scaler, scheduler)
  File "/workspace/MVDeTr/multiview_detector/trainer.py", line 51, in train
    (world_heatmap, world_offset), (imgs_heatmap, imgs_offset, imgs_wh) = self.model(data, affine_mats)
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/modules/module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/workspace/MVDeTr/multiview_detector/models/mvdetr.py", line 202, in forward
    world_feat = self.world_feat(world_feat, visualize=visualize)
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/modules/module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/workspace/MVDeTr/multiview_detector/models/trans_world_feat.py", line 98, in forward
    memory = self.encoder(src_flatten, spatial_shapes, level_start_index, valid_ratios, lvl_pos_embed_flatten)
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/modules/module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/workspace/MVDeTr/multiview_detector/models/deformable_transformer.py", line 50, in forward
    output = layer(output, pos, reference_points, spatial_shapes, level_start_index, padding_mask)
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/modules/module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/workspace/MVDeTr/multiview_detector/models/deformable_transformer.py", line 77, in forward
    src2 = self.self_attn(self.with_pos_embed(src, pos), reference_points, src, spatial_shapes, level_start_index,
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/modules/module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/workspace/MVDeTr/multiview_detector/models/ops/modules/ms_deform_attn.py", line 104, in forward
    sampling_locations = reference_points[:, :, None, :, None, :] \
RuntimeError: The size of tensor a (7) must match the size of tensor b (8) at non-singleton dimension 3
```

**Comment by hou-yz** [comment]
Please check the bug fix in commit 87783a7f7dcf77ea96760e4c49478c151121cd0e

Best,
Yunzhong

**Comment by luorix1** [comment]
Thank you for the speedy bug fix!
